test/iterators.jl

```diff
@@ -1074,6 +1074,30 @@ end
     end
 end
 
+@testset "nth" begin
+    Z = Array{Int,0}(undef)
+    Z[] = 17
+    itrs = (collect(1:1000), 10:6:1000, "∀ϵ>0", (1, 3, 5, 10, 78), reshape(1:30, (5, 6)),
```

Comment:
I would put these in a `Vector{Any}` - no need to create specialized code for long tuples here. Since the test only runs once, using a tuple will be much slower.

Reply:
This made the test 10x faster with way less allocations (300+ to 5), guess the compiler was chocking on the overcomplicated tuples.